File: languages/js/src/entity/identity_v2/integer_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { AttributeDescriptor } from "./attribute_descriptor";

export class IntegerAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.IntegerAttributeDefinitionV2 {
    return identityEntitiesProto.IntegerAttributeDefinitionV2.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.IntegerAttributeDefinitionV2
  ): IntegerAttributeDescriptor {
    return new IntegerAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}

File: languages/python/bloock/entity/identity_v2/integer_attribute_descriptor.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute_descriptor import AttributeDescriptor


class IntegerAttributeDescriptor(AttributeDescriptor):
    @staticmethod
    def from_proto(a: proto.IntegerAttributeDefinitionV2) -> IntegerAttributeDescriptor:
        return IntegerAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
            required=a.required,
        )

    def to_proto(self) -> proto.IntegerAttributeDefinitionV2:
        return proto.IntegerAttributeDefinitionV2(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
            required=self.required,
        )